spec/app/domain/authentication/resource_restrictions/extract_resource_restrictions_spec.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe(Authentication::ResourceRestrictions::ExtractResourceRestrictions) do
+  let(:authenticator_name) { 'authn-name' }
+  let(:service_id) { 'service-id' }
+  let(:role_name) { 'role-name' }
+  let(:account) { 'account' }
+
+  let(:mocked_fetch_resource_annotations_instance_invalid) { double("FetchResourceAnnotationsInvalid") }
+  let(:fetch_resource_annotations_instance_error) { "FetchResourceAnnotationsError" }
+  let(:mocked_fetch_resource_annotations_instance_empty) { double("FetchResourceAnnotationsEmpty") }
+  let(:mocked_fetch_resource_annotations_instance_valid) { double("FetchResourceAnnotationsValid") }
+
+
+  let(:resource_restrictions_empty_hash) {
+    Authentication::ResourceRestrictions::ResourceRestrictions.new(
+      resource_restrictions_hash: {}
+    )
+  }
+
+  before(:each) do
+    allow(mocked_fetch_resource_annotations_instance_invalid).to(
+      receive(:call).and_raise(fetch_resource_annotations_instance_error)
+    )
+
+    allow(mocked_fetch_resource_annotations_instance_empty).to(
+      receive(:call).and_return({})
+    )
+
+    allow(mocked_fetch_resource_annotations_instance_valid).to(
+      receive(:call).and_return(
+        "#{authenticator_name}/#{service_id}/claim1" => "claim1-value",
+        "#{authenticator_name}/#{service_id}/claim2" => "claim2-value"
```

Comment:
Do you want to add annotation belongs to other service id and to validate that it's ignored
And to add not related annotation at all

Reply:
@sashaCher This is the responsibility of this class `GetServiceSpecificRestrictionFromAnnotation` and we already have UT for it

Reply:
From one side you're right.
From other you do not mock it so the tests are running the original class behavior and this is the place where UT starting to look a bit more "integrationnal". Means we can catch if the way we're integrating classes together is right.
CC: @eladkug